### ISSUE TITLE
System.ArgumentNullException while opening .editorconfig file

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#53831]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/SystemArgumentNullException-while-openi/1435353)._

---
[severity:It bothers me. A fix would be nice]
When trying to open a .editorconfig file in one of my projects, I received a System.ArgumentNullException exception:

```
An error occurred while initializing the frame's content
System.ArgumentNullException: Value cannot be null.
Parameter name: path
   at System.IO.DirectoryInfo.. ctor(String path)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.<>c.<GetProjectsForPath>b__0_1(Project p)
   at System.Linq.Enumerable.<>c__DisplayClass7_0`3. <CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.GetProjectsForPath(Solution solution, String givenPath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.SettingsProviderBase`4.Update()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.Formatting.CommonFormattingSettingsProviderFactory.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.CombinedOptionsProviderFactory`1.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.SettingsAggregator.GetSettingsProvider[TData](String fileName)
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane. <Initialize>g__GetFormattingView|12_0()
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.Initialize()
   at Microsoft.VisualStudio.Shell.WindowPane.InternalSetSite(IServiceProvider p)
   at Microsoft.VisualStudio.Shell.WindowPane.Microsoft.VisualStudio.Shell.Interop.IVsWindowPane.SetSite(IServiceProvider psp)
   at Microsoft.VisualStudio.Shell.WindowPane.Microsoft.VisualStudio.Shell.Interop.IVsUIElementPane.SetUIElementSite(IServiceProvider p)
   at Microsoft.VisualStudio.Platform.WindowManagement.UIElementDocumentObject.SetSite(DocumentObjectSite site)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.InitializeDocumentObject(Object punkView)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.InitializeDocumentSite(Boolean creatingStubFrame, Boolean replacingStubView, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowManagerService.CreateContentPane(FrameMoniker frameMoniker, Boolean isDocument, String lpstrMkDoc, UInt32 eCreateWindowFlags, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid, Guid rguidCmdUI, ViewGroup parent, IVsWindowFrame& ppWindowFrame)

```

---
#### Feedback Bot on 5/28/2021, 04:05 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Feedback Bot on 5/28/2021, 10:38 PM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

**Comment by Bosch-Eli-Black** [comment]
Thanks 🙂  Original error report was by me.

**Comment by jmarolf** [comment]
Cannot reproduce I believe this was already fixed

**Comment by timritzer** [comment]
@jmarolf Can still recreate this one. Recreation is pretty simple, have your editor config in a parent folder to the solution, and included in the solution.

Example:
Repo Root
    .editorconfig
    Solution1
    Solution2

This will apply to both solutions, and is supported. However the editor will blow up when trying to edit it.

Moving it into the solution folder with the exact same content makes the error go away.

This is problematic for mono-repo's with consistent coding standards.
```
An error occurred while initializing the frame's content
System.ArgumentNullException: Value cannot be null.
Parameter name: path
   at System.IO.DirectoryInfo..ctor(String path)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.<>c.<GetProjectsForPath>b__0_1(Project p)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.GetProjectsForPath(Solution solution, String givenPath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.SettingsProviderBase`4.Update()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.Formatting.CommonFormattingSettingsProviderFactory.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.CombinedOptionsProviderFactory`1.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.SettingsAggregator.GetSettingsProvider[TData](String fileName)
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.<Initialize>g__GetFormattingView|12_0()
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.Initialize()
   at Microsoft.VisualStudio.Shell.WindowPane.InternalSetSite(IServiceProvider p)
   at Microsoft.VisualStudio.Platform.WindowManagement.UIElementDocumentObject.SetSite(DocumentObjectSite site)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.InitializeDocumentObject(Object punkView)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.InitializeDocumentSite(Boolean creatingStubFrame, Boolean replacingStubView, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowManagerService.CreateContentPane(FrameMoniker frameMoniker, Boolean isDocument, String lpstrMkDoc, UInt32 eCreateWindowFlags, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid, Guid rguidCmdUI, ViewGroup parent, IVsWindowFrame& ppWindowFrame)
```

**Comment by jmarolf** [comment]
Confirmed this does not reproduce on the latest builds

**Comment by syska** [comment]
Latest stable: `17.0.4` Microsoft Visual Studio Community 2022 (64-bit)
```
An error occurred while initializing the frame's content
System.ArgumentNullException: Value cannot be null.
Parameter name: path
   at System.IO.DirectoryInfo..ctor(String path)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.<>c.<GetProjectsForPath>b__0_1(Project p)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.GetProjectsForPath(Solution solution, String givenPath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.SettingsProviderBase`4.Update()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.Whitespace.CommonWhitespaceSettingsProviderFactory.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.CombinedOptionsProviderFactory`1.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.SettingsAggregator.GetSettingsProvider[TData](String fileName)
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.<>c__DisplayClass12_0.<Initialize>g__GetWhitespaceView|1()
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.Initialize()
   at Microsoft.VisualStudio.Shell.WindowPane.InternalSetSite(IServiceProvider p)
   at Microsoft.VisualStudio.Platform.WindowManagement.UIElementDocumentObject.SetSite(DocumentObjectSite site)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.InitializeDocumentObject(Object punkView)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.InitializeDocumentSite(Boolean creatingStubFrame, Boolean replacingStubView, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowManagerService.CreateContentPane(FrameMoniker frameMoniker, Boolean isDocument, String lpstrMkDoc, UInt32 eCreateWindowFlags, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid, Guid rguidCmdUI, ViewGroup parent, IVsWindowFrame& ppWindowFrame)

```

Solution is in: `C:\Projects\cadesign\Vipp`

It's in a solutuin folder called: `.rules` ... 

The file is located here: `C:\Projects\cadesign\Vipp\src\.editorconfig` ...

**Comment by syska** [comment]
Latest stable: `17.0.4` Microsoft Visual Studio Community 2022 (64-bit)
```
An error occurred while initializing the frame's content
System.ArgumentNullException: Value cannot be null.
Parameter name: path
   at System.IO.DirectoryInfo..ctor(String path)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.<>c.<GetProjectsForPath>b__0_1(Project p)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.GetProjectsForPath(Solution solution, String givenPath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.SettingsProviderBase`4.Update()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.Whitespace.CommonWhitespaceSettingsProviderFactory.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.CombinedOptionsProviderFactory`1.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.SettingsAggregator.GetSettingsProvider[TData](String fileName)
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.<>c__DisplayClass12_0.<Initialize>g__GetWhitespaceView|1()
   at Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane.Initialize()
   at Microsoft.VisualStudio.Shell.WindowPane.InternalSetSite(IServiceProvider p)
   at Microsoft.VisualStudio.Platform.WindowManagement.UIElementDocumentObject.SetSite(DocumentObjectSite site)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.InitializeDocumentObject(Object punkView)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.InitializeDocumentSite(Boolean creatingStubFrame, Boolean replacingStubView, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowManagerService.CreateContentPane(FrameMoniker frameMoniker, Boolean isDocument, String lpstrMkDoc, UInt32 eCreateWindowFlags, Object punkView, Object punkData, IServiceProvider pServiceProvider, IVsUIHierarchy pUIHierarchy, UInt32 vsid, Guid rguidCmdUI, ViewGroup parent, IVsWindowFrame& ppWindowFrame)

```

Solution is in: `C:\Projects\syska\project`

It's in a solutuin folder called: `.rules` ... 

The file is located here: `C:\Projects\syska\project\src\.editorconfig` ...

**Comment by vlade-rc** [comment]
why this is closed without solution, I have the same issue with 17.0.4

**Comment by syska** [comment]
> why this is closed without solution, I have the same issue with 17.0.4

I really have no idea ... it also fails on `17.0.5` ... so if any MS people are listening here. Please reach out if you need more information about the issue.

**Comment by Bosch-Eli-Black** [comment]
@jmarolf, Should this be reopened? Seems people are still having this issue (although it's been fixed for me 🙂).

**Comment by Ridlsan** [comment]
Same here

**Comment by jmarolf** [comment]
Thanks folks, sorry that this is still happening. I did get another hit on this callstack that I am investigating. I am tracking that in a new issue here: https://github.com/dotnet/roslyn/issues/58732. If you have any more information to add please put it over there.

**Comment by syska** [comment]
> Thanks folks, sorry that this is still happening. I did get another hit on this callstack that I am investigating. I am tracking that in a new issue here: #58732. If you have any more information to add please put it over there.

I can't really figure WHY ... because if I start moving the file around and add it again. Sometimes it starts to work.

So then I think it might be a User Context issue ... but having no knowledge of how VS works ... I not sure what to try. But better exception message about what exactly is missing or can't be found would be good.

I will update the other issues if I find anything new.

**Comment by csharpforevermore** [comment]
I get it as well (Visual Studio Community (64-bit v17.0.2) in a .NET Framework solution (MVC - v4.7.2).  I left click the project so it is highlighted, right click the project, select "Add" > "New Item" and then I searched for EditorConfig.  I get 2 options (.NET) and (default). I select .NET.  The file name is ".editorconfig". When I click the "Add" button, I get a popup window titled "Microsoft Visual Studio" with a red cross on the left, and a message "Value cannot be null. Parameter name: path" with a single "OK" button.

This seems to occur anywhere I try to create a file of this type (any folder), and happens with both types of EditorConfig file (.NET and default).

Strangely, the file still gets created.

**Comment by csharpforevermore** [comment]
Also if i try to open the file, it throws the same exception. Here's the stacktrace message

 error occurred while initializing the frame's content
System.ArgumentNullException: Value cannot be null.
Parameter name: path
   at System.IO.DirectoryInfo..ctor(String path)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.<>c.<GetProjectsForPath>b__0_1(Project p)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Extensions.SolutionExtensions.GetProjectsForPath(Solution solution, String givenPath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.SettingsProviderBase`4.Update()
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.Whitespace.CommonWhitespaceSettingsProviderFactory.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.DataProvider.CombinedOptionsProviderFactory`1.GetForFile(String filePath)
   at Microsoft.CodeAnalysis.Editor.EditorConfigSettings.SettingsAggregator.GetSettingsProvider[TData](String fileName)

File Path: <private>\.editorconfig
Frame GUID: 68b46364-d378-42f2-9e72-37d8gc5f4468
Frame mode: VSFM_Float
Error code: 0x80004003

**Comment by jmarolf** [comment]
Taking a look the fix for this is in 17.1 which has not been released yet. You can install the preview to get it early
